### DISCUSSION TITLE
Icelandic - Two hours of recording [VITS]

### DISCUSSION THREAD

**Discussion by kksteini** [coqui-ai/TTS#1627] [Show and tell]
I'm pleased about how the quality is steadily improving with more data. I'm recording around 11 minutes of audio on the days when I record and am eager to hit my goal of 6 hours and then try training again.

Here is a random sentence I thought of generating and then I recorded it again myself for comparison.

### Góðan daginn. Þú hefur náð sambandi við Júní. Vinsamlegast skildu eftir skilaboð.
https://user-images.githubusercontent.com/77672665/172325452-b6fe47d4-cfe0-4494-9244-00a253a2a10b.mp4

and then a real recording of the same text for comparison (made this morning so a bit different morning voice)

https://user-images.githubusercontent.com/77672665/172325734-32a4f186-4bab-4007-ba71-9a7a43d15236.mp4

--- 

Then for fun I tried sounding out an english sentence with Icelandic :grin: 

### Gúdd morning kókví tí tí ess. Ðiss iss mæ pró-gress öpp deit. Æ træd svitsing tú ðe ví æ tí ess model end itt has somm næs intermídíerí átpútt.

_Good morning Coqui TTS. This is my progress update. I tried switching to the VITS model and it has some nice intermediary output_

https://user-images.githubusercontent.com/77672665/172325996-ec646790-1f44-43fc-8057-cd931abf02d5.mp4

**Comment by p0p4k** [comment]
Looks good.

**Comment by erogol** [comment]
awesome!

**Comment by erogol** [comment]
If you want we can add your model to https://huggingface.co/spaces/coqui/CoquiTTS when it is done.

**Reply by kksteini** [reply]
I might be interested.
On one hand I don't know how I feel about anyone being able to generate my voice but on the other hand I am myself recording because of lack of data.

There are no freely available TTSs that I'm aware.

I'll have to give this some thought

**Comment by Rico2000** [comment]
Hey,
nice work! Can you share the config you used for Training?

**Reply by kksteini** [reply]
https://github.com/coqui-ai/TTS/blob/dev/recipes/ljspeech/vits_tts/train_vits.py
I used this recipe except I changed the config as such

```python
config = VitsConfig(
    audio=audio_config,
    run_name="vits_ljspeech",
    batch_size=16,                         #changed
    eval_batch_size=8,                     #changed
    batch_group_size=2,                    #changed
    num_loader_workers=0,
    num_eval_loader_workers=2,             #changed
    run_eval=True,
    test_delay_epochs=-1,
    epochs=1000,
    text_cleaner="phoneme_cleaners",       #changed
    use_phonemes=True,
    phoneme_language="is",                 #changed
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    compute_input_seq_cache=True,
    print_step=25,
    print_eval=True,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
)```

**Reply by kksteini** [reply]
I'm skirting around the edges with this having around 8GB VRAM.
I also had to edit the parameter responsible for the max audio length to exclude auio files longer than eight and a half seconds but this can also be done by hand.
